my_init.d/factory_reset.sh

```diff
@@ -1,34 +1,50 @@
-#!/bin/bash -e
-
-# Debugging.
-set -x
+#!/bin/bash -eu
 
 # Environment.
 export SHELL=/bin/bash
 
-# Performing factory reset of your AiiDAlab environment:
-# 0 - No reset (noop).
-# 1 - Remove locally installed software and apps (removes ~/apps/ and ~/.local/).
-# 2 - Remove all files and directories within the users home directory.
-
-
-case "${AIIDALAB_FACTORY_RESET}" in
-
-  0)
-    exit 0
-    ;;
-
-  1)
-    rm -rf "/home/${SYSTEM_USER}/apps"
-    rm -rf "/home/${SYSTEM_USER}/.local"
-    ;;
-
-  2)
-    find /home/${SYSTEM_USER}/ -mindepth 1 -delete
-    ;;
-
-  *)
-    echo "Unknown factory reset mode: ${AIIDALAB_FACTORY_RESET}"
-    exit 1
-    ;;
-esac
+# Bit mask for performing factory reset of the AiiDAlab environment:
+# 0b000 0 - No reset (noop).
+# 0b001 1 - Remove locally installed software and apps (removes ~/apps/ and ~/.local/).
+# 0b010 2 - Remove all files and directories within the users home directory.
+# 0b100 4 - Remove the factory reset file after performing reset.
+
+# First try to read the reset mode from an environment variable.
+RESET_MODE="${AIIDALAB_FACTORY_RESET:-}"
+
+# If the mode was not set via environment variable, attempt to read from file.
+if [ -z ${RESET_MODE} ] && [ -e "${HOME}/AIIDALAB_FACTORY_RESET" ]; then
+  RESET_MODE="`cat ${HOME}/AIIDALAB_FACTORY_RESET`"
+fi
+
+# If the mode was not set via environment variable or file, assume mode=0.
+if [ -z ${RESET_MODE} ]; then
+  echo "No factory reset mode set, assuming mode=0 (no reset)."
+  RESET_MODE=0
+fi
+
+# Perform the individual reset functions:
+
+# mode==0 -> noop
+if (( ${RESET_MODE} == 0)); then  # noop
+    echo "Factory reset: No reset (mode=0)."
+fi
+
+# mode & 001 -> delete ~/apps/ and ~/.local/
+if (( (${RESET_MODE} & 0x1) == 0x1)); then
+  echo "Factory reset: Remove apps (~/apps/) and local software installation (~/.local/)."
+  rm -rf "/home/${SYSTEM_USER}/apps"
+  rm -rf "/home/${SYSTEM_USER}/.local"
+fi
+
+# mode & 010 -> delete all home directory contents
+if (( (${RESET_MODE} & 0x2) == 0x2)); then
+  echo "Factory reset: Remove user home directory contents."
+  find /home/${SYSTEM_USER}/ -mindepth 1 -delete
```

Comment:
```suggestion
  find "/home/${SYSTEM_USER}/" -mindepth 1 -delete
```
I believe this should work, but please test.

Reply:
Done.